VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx

```diff
@@ -0,0 +1,364 @@
+
+            break;
+          case vr::TrackingResult_Calibrating_OutOfRange:
+            d->pushButton_Transform->setIcon(QIcon(":/Icons/headset_status_range.png"));
+            break;
+          default:
+            d->pushButton_Transform->setIcon(QIcon(":/Icons/headset_status_off.png"));
+        }
+        break;
+      }
+    case vr::TrackedDeviceClass_GenericTracker:
+      {
+        if (!d->VRViewNode->GetTrackerTransformUpdate())
+        {
+          d->pushButton_Transform->setIcon(QIcon(":/Icons/tracker_status_off.png"));
+          break;
+        }
+        switch (status)
+        {
+          case vr::TrackingResult_Running_OK:
+            d->pushButton_Transform->setIcon(QIcon(":/Icons/tracker_status_ready.png"));
+
+            break;
+          case vr::TrackingResult_Running_OutOfRange:
+            d->pushButton_Transform->setIcon(QIcon(":/Icons/tracker_status_range.png"));
+
+            break;
+          case vr::TrackingResult_Calibrating_InProgress:
+            d->pushButton_Transform->setIcon(QIcon(":/Icons/tracker_status_alert.png"));
+
+            break;
+          case vr::TrackingResult_Calibrating_OutOfRange:
+            d->pushButton_Transform->setIcon(QIcon(":/Icons/tracker_status_range.png"));
+
+            break;
+          default:
+            d->pushButton_Transform->setIcon(QIcon(":/Icons/tracker_status_off.png"));
+
+        }
+        break;
+      }
+    case vr::TrackedDeviceClass_Controller:
+      {
+        if (!d->VRViewNode->GetControllerTransformsUpdate())
+        {
+          d->pushButton_Transform->setIcon(QIcon(":/Icons/controller_status_off.png"));
+          break;
+        }
+        switch (status)
+        {
+          case vr::TrackingResult_Running_OK:
+            d->pushButton_Transform->setIcon(QIcon(":/Icons/controller_status_ready.png"));
+
+            break;
+          case vr::TrackingResult_Running_OutOfRange:
+            d->pushButton_Transform->setIcon(QIcon(":/Icons/controller_status_range.png"));
+
+            break;
+          case vr::TrackingResult_Calibrating_InProgress:
+            d->pushButton_Transform->setIcon(QIcon(":/Icons/controller_status_alert.png"));
+
+            break;
+          case vr::TrackingResult_Calibrating_OutOfRange:
+            d->pushButton_Transform->setIcon(QIcon(":/Icons/controller_status_range.png"));
+
+            break;
+          default:
+            d->pushButton_Transform->setIcon(QIcon(":/Icons/controller_status_off.png"));
+
+        }
+      }
+      break;
+  }
+}
+
+//-----------------------------------------------------------------------------
+void qMRMLVirtualRealityTransformWidget::findViewNode()
+{
+  Q_D(qMRMLVirtualRealityTransformWidget);
+
+  if (d->VRViewNode == nullptr)
+  {
+    // Get access to the VR subsystem
+    for (QWidget* widget : qSlicerApplication::application()->topLevelWidgets())
+    {
+      if (widget->objectName().compare("VirtualRealityWidget") == 0)
```

Comment:
This is a bit of a hack. Is it possible to set it somehow from outside, or make it accessible without traversing widgets and comparing object names?

Reply:
Definitely, I hate it. But the only other way is to access it via the module class, and you can't access the module library from the widget library.

Reply:
Maybe the view node could be passed into the widget constructor? That's not bad...